src/mongocxx/events/command_failed_event.hpp

```diff
@@ -75,6 +79,15 @@ class MONGOCXX_API command_failed_event {
     ///
     std::int64_t operation_id() const;
 
+    ///
+    /// Returns the service id.
+    ///
+    /// @note service_id() only returns a value if load balancing is enabled.
+    ///
+    /// @return No value, or the service id if load balancing is enabled.
```

Comment:
Nit: other documentation in the code base refers to returns with optionals as possibly returning a "disengaged" value. Here is one example: https://github.com/kevinAlbs/mongo-cxx-driver/blob/33416f88f1d59e19eb103fedab59d73399332a7c/src/mongocxx/collection.hpp#L262:L262

"disengaged" always sounded odd to me, but I would consider changing for consistency.

Reply:
The C++ standard does not use this terminology (n4892, 20.6.2), instead using "contained object" and "contains a value". Perhaps the phrasing "@return No contained value, or contains the service id..." is the way to go. I feel that perpetuating "disengaged" might continue to add further confusion to the mix.